### PR TITLE
Update documentation for prost remote plugins

### PR DESCRIPTION
**Pull request by pkwarren**
These plugins are now maintained and automatically updated by the buf team at https://github.com/bufbuild/plugins. Update the documentation to point to the new plugin locations and use the new syntax to invoke remote plugins, as the previous plugins alpha is deprecated and going away on July 10th.

### CODE REVIEW

**Comment by pkwarren** [comment]
I noticed there is also a `publish_buf.sh` which can be removed from this project (I don't see references to it any longer, but it won't work any more). Updates to plugins hosted by Buf on the BSR should be made upstream at https://github.com/bufbuild/plugins.

**Comment by pkwarren** [comment]
This can replace the work in #66 as the crate plugin is also now available.

File: publish_buf.sh [hunk 1]
```diff
@@ -1,18 +0,0 @@
...
-set -euxo pipefail
-
-ITER=${1:-1}
-
-buildImage() {
-    PLUGIN=$1
-    IMAGE=$2
-    VERSION=$(cargo read-manifest --manifest-path "protoc-gen-$PLUGIN/Cargo.toml" 2>/dev/null | jq -r .version)
-
-    docker buildx build --platform=linux/amd64 --build-arg "BIN=protoc-gen-$PLUGIN" -t "plugins.buf.build/prost/$IMAGE:$VERSION-$ITER" --push .
```
Comment (pkwarren) [inline publish_buf.sh:12]: FYI the `plugins.buf.build` domain is going away with the alpha remote removal. This script will no longer be functional so it should be safe to remove.

**Comment by pkwarren** [comment]
@neoeinstein - I'm not sure if you're actively working on this project any more, but these changes would be good to land prior to July 10th - that's the date we're removing support for the prior alpha - see https://buf.build/docs/migration-guides/migrate-remote-generation-alpha/ for more details.

**Comment by neoeinstein** [comment]
Hey @pkwarren, Sorry for the delay; something seems to have black-holed my GitHub notifications, but I'm working on fixing that. I'm reviewing my backlog here now, and will get stuff prepared shortly.

**Review by neoeinstein — APPROVED** [review]

**Pull request merged**